src/pages/LogOutPreviousUserPage.tsx

```diff
@@ -42,18 +50,28 @@ function LogOutPreviousUserPage({session, route}: LogOutPreviousUserPageProps) {
                 const shortLivedAuthToken = route.params.shortLivedAuthToken ?? '';
                 SessionActions.signInWithShortLivedAuthToken(email, shortLivedAuthToken);
             }
+            const exitTo = route.params.exitTo as Route | null;
+            // We don't want to navigate to the exitTo route when creating a new workspace from a deep link,
+            // because we already handle creating the optimistic policy and navigating to it in App.setUpPoliciesAndNavigate,
+            // which is already called when AuthScreens mounts.
+            if (exitTo && exitTo !== ROUTES.WORKSPACE_NEW && !account?.isLoading && !isLoggingInAsNewUser) {
+                Navigation.isNavigationReady().then(() => {
+                    // remove this screen and navigate to exit route
+                    const exitUrl = NativeModules.HybridAppModule ? Navigation.parseHybridAppUrl(exitTo) : exitTo;
+                    Navigation.goBack();
+                    Navigation.navigate(exitUrl);
+                });
+            }
         });
-
-        // We only want to run this effect once on mount (when the page first loads after transitioning from OldDot)
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+    }, [initUrl, account, route, session]);
```

Comment:
Are you sure we want this to run if any of these change? I'd think that this might be a case to suppress the lint rule and pass an empty URL. I imagine `initUrl` will always be defined at this point if it will be because it's available during the first render of the top-level App component. By the time we get down to this point in the tree, I imagine the context will be "hydrated" with that value.